src/main/java/com/byclosure/webcat/stepdefinitions/helpers/SeleniumInteractions.java

```diff
@@ -0,0 +1,192 @@
+package com.byclosure.webcat.stepdefinitions.helpers;
+
+import com.byclosure.webcat.context.IContext;
+import org.openqa.selenium.*;
+import org.openqa.selenium.internal.selenesedriver.TakeScreenshot;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class SeleniumInteractions {
+
+    private final WebDriver driver;
+    private final IContext context;
+
+    public SeleniumInteractions(WebDriver driver, IContext context) {
+        this.driver = driver;
+        this.context = context;
+    }
+
+    public void clickLink(String locator) {
+        WebElement linkElement = findLinkElement(locator);
+        linkElement.click();
+    }
+    
+    public void clickButton(String locator){
+        WebElement buttonElement = findButtonElement(locator);
+        buttonElement.click();
+    }
+
+    public boolean searchForContent(String text) {
+        final WebElement body = driver.findElement(By.xpath("//*"));
+        return body.isDisplayed() && body.getText().contains(text);
+    }
+    
+    public void fillFormTextField(String field, String value){
+        final WebElement elementById = findElementById(field);
+        final WebElement elementByName = findElementByName(field);
+        final WebElement elementByText = findInputByText(field);
+
+        if(elementById != null) {
+            elementById.sendKeys(value);
+        } else if(elementByName != null) {
+            elementByName.sendKeys(value);
+        } else if(elementByText != null) {
+            elementByText.sendKeys(value);
+        } else {
+            throw new NoSuchElementException("\"" + field + "\" not found.");
+        }
+    }
+    
+    public boolean formFieldContains(String field, String parent, String value) {
+        final String parentSelector; 
+        if(null == parent || "".equals(parent)) {
+            parentSelector = "form";
+        } else {
+            parentSelector = parent;
+        }
+        
+        final WebElement parentElement = driver.findElement(By.cssSelector(parentSelector));
+        final WebElement element = parentElement.findElement(By.cssSelector(field));
+        final String elementValue = element.getText();
+        
+        return elementValue.contains(value);
+    }
+
+    public void takeScreenshot() {
+        if(driver instanceof TakeScreenshot) {
+            final byte[] screenshot = ((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES);
+            context.addScreenshot(screenshot);
+        }
+    }
+
+    private WebElement findLinkElement(String locator) {
```

Comment:
Please rely in the existing web driver methods. We do not need to introduce new locators.